### PR TITLE
RST backtick refactor (all *.rst except manual.rst and rst_examples.rst)

### PR DESCRIPTION
**Pull request by quantimnot**
Added `.. default-role:: code` directive to all rst files except manual.rst.

Replaced double backticks with single backticks and realign shifted table columns in all rst files except manual.rst.

Replaced triple backtick code blocks with `.. code-block:: nim` code blocks.

### CODE REVIEW

**Comment by timotheecour** [comment]
* followup https://github.com/nim-lang/Nim/pull/17028
* check if this is affected by https://github.com/nim-lang/Nim/issues/17260, if so we may need to wait until that bug is fixed

File: doc/docstyle.rst
Comment (timotheecour) [inline doc/docstyle.rst:18]: bad change here

File: doc/manual_experimental_strictnotnil.rst
Comment (timotheecour) [inline doc/manual_experimental_strictnotnil.rst:146]: pre-existing, but renders badly because of the space that prevents interpretation

File: doc/tut1.rst
Comment (timotheecour) [inline doc/tut1.rst:107]: bad change; should be fixed in followup PR

File: doc/tut1.rst
Comment (timotheecour) [inline doc/tut1.rst:117]: ditto

File: doc/tut1.rst
Comment (timotheecour) [inline doc/tut1.rst:816]: ditto

File: doc/tut1.rst
Comment (timotheecour) [inline doc/tut1.rst:823]: ditto

**Review by timotheecour — APPROVED** [review]
LGTM, I checked everything using the same technique as described in https://github.com/nim-lang/Nim/pull/17259#pullrequestreview-615964357:
checking both the diff, as well as the rendered html files and checking that each difference (a very small number) is intentional, or noting un-intentional ones as TODO for a followup PR

**Comment by timotheecour** [comment]
merged, same rationale as https://github.com/nim-lang/Nim/pull/17259#issuecomment-802399932

@quantimnot thanks for this and the other PR! could you please followup with a PR to address the above mentioned points?

also, a separate PR welcome for finishing off the work started here: https://github.com/nim-lang/Nim/pull/17217 so we can finally turn the page on this :)

**Comment by quantimnot** [comment]
Yeah, I'll try to follow up on this pr today or tomorrow.
I need to pass on #17217 until I have some other work finished.

OT: The other work is GDB/LLDB/... extensions for Nim. I remember reading a comment from you regarding some
work you did for a LLDB plugin. Do you have any work (even sloppy experimental code) that you can share? I'd primarily be interested in any formatters you have developed.

**Comment by quantimnot** [comment]
I'm going to wait until #17315 is merged.

**Comment by timotheecour** [comment]
@quantimnot https://github.com/nim-lang/Nim/pull/17315 has been merged, you can go ahead with followups for this PR as well as https://github.com/nim-lang/Nim/pull/17259

> OT: The other work is GDB/LLDB/.

followup https://github.com/timotheecour/Nim/issues/599#issuecomment-806053450

**Pull request merged**
